graylog2-web-interface/src/components/common/InteractableModal.jsx

```diff
@@ -0,0 +1,197 @@
+
+    const tooFarUp = y < -height;
+    const tooFarDown = y > document.body.offsetHeight + height;
+    const newDown = tooFarDown ? document.body.offsetHeight - height : y;
+    const newY = tooFarUp ? 0 : newDown;
+
+    const setPosition = {
+      x: newX,
+      y: newY,
+    };
+    setDragPosition(setPosition);
+    onDrag(setPosition);
+  };
+
+  return (
+    <InteractableModalWrapper className={wrapperClassName}>
+      <StyledRnd default={{ ...position, ...size }}
+                 minHeight={minHeight}
+                 minWidth={minWidth}
+                 maxHeight={document.body.offsetHeight}
+                 maxWidth={document.body.offsetWidth}
+                 dragHandleClassName={dragHandleClassName}
+                 onDragStop={handleDragStop}
+                 onResizeStop={(event, direction, ref) => {
+                   const newSize = {
+                     width: ref.style.width,
+                     height: ref.style.height,
+                   };
+                   setResizeSize(newSize);
+                   onResize(newSize);
+                 }}
+                 position={dragPosition}
+                 size={resizeSize}
+                 className={className}>
+        <Header ref={dragHandleRef}>
+          <Title><DragBars name="bars" />{' '}{title}</Title>
+
+          <Button bsStyle="default" onClick={onClose} bsSize="sm">
+            <Icon name="times" size="lg" />
+          </Button>
+        </Header>
+
+        <Content>
+          {children}
+        </Content>
+      </StyledRnd>
+    </InteractableModalWrapper>
+  );
+};
+
+InteractableModal.propTypes = {
+  /** className that will be applied to `react-rnd` */
+  className: PropTypes.string,
+  /** Content of the InteractableModal modal */
+  children: PropTypes.node.isRequired,
+  /** Minimum height that modal can be reduced to */
+  minHeight: PropTypes.number,
+  /** Minimum width that modal can be reduced to */
+  minWidth: PropTypes.number,
+  /** Function that is called when InteractableModal is closed */
+  onClose: PropTypes.func,
+  /** Function that is called when InteractableModal has finished being dragged */
+  onDrag: PropTypes.func,
+  /** Function that is called when InteractableModal has finished being resized */
+  onResize: PropTypes.func,
+  /** If you want to control InteractableModal you can pass specific position */
+  position: PropTypes.shape({
+    x: PropTypes.number,
+    y: PropTypes.number,
+  }),
+  /** If you want to control InteractableModal you can pass specific size */
+  size: PropTypes.shape({
+    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  }),
+  /** Title that appears at the top of the window */
+  title: PropTypes.string,
+  /** Used to style wrapping component */
+  wrapperClassName: PropTypes.string,
+};
+
+InteractableModal.defaultProps = {
+  className: undefined,
+  minHeight: 250,
+  minWidth: 250,
```

Comment:
Resizing to the minimum values look odd when the warning is not dismissed:

<img width="490" alt="Screenshot 2019-12-05 at 12 32 40" src="https://user-images.githubusercontent.com/716185/70232326-783e4b80-175c-11ea-930a-477ac948a52e.png">
<img width="275" alt="Screenshot 2019-12-05 at 12 32 49" src="https://user-images.githubusercontent.com/716185/70232328-783e4b80-175c-11ea-9517-00578b70c0a7.png">

Once the warning is gone, everything looks good again.